sql/core/src/main/scala/org/apache/spark/sql/SQLConf.scala

```diff
@@ -93,7 +94,7 @@ private[spark] object SQLConf {
           isPublic: Boolean = true): SQLConfEntry[Int] =
       SQLConfEntry(key, defaultValue, { v =>
         try {
-          v.toInt
+          Utils.byteStringAsBytes(v).toInt
```

Comment:
It looks like you're making all integer config values parse as if they're memory sizes. That's not right, right?